compiler/x/codegen/FPTreeEvaluator.cpp

```diff
@@ -1040,11 +1040,15 @@ TR::Register *OMR::X86::TreeEvaluator::fpConvertToInt(TR::Node *node, TR::Symbol
    startLabel->setStartInternalControlFlow();
    reStartLabel->setEndInternalControlFlow();
 
+
+   TR_ASSERT(cg->comp()->target().cpu.supportsFeature(OMR_FEATURE_X86_SSE) == cg->getX86ProcessorInfo().supportsSSE(), "supportsSSE() failed\n");
+   TR_ASSERT(cg->comp()->target().cpu.supportsFeature(OMR_FEATURE_X86_SSE2) == cg->getX86ProcessorInfo().supportsSSE2(), "supportsSSE2() failed\n");
```

Comment:
Should these be fatal? Similar comment for all other related asserts. Also I assume eventually the APIs on the right hand side of the equality will be going away, right?

Reply:
Okay I will make them back to fatal. Originally I added them as fatal for my own testing. I later changed them to normal asserts for the PR as I'm not sure how much overhead this will add to the system.

Reply:
If we are going to remove the "old" APIs once we can initialize the port library properly then the asserts will be short lived. I'd be ok with taking a penalty to ensure functional correctness while in this transition stage.